src/main/java/org/commcare/formplayer/util/PrototypeUtils.java

```diff
@@ -138,6 +140,8 @@ public static void setupThreadLocalPrototypes() {
                 BubbleSeries.class.getName(),
                 SendAction.class.getName(),
                 XPathDistinctValuesFunc.class.getName(),
+                ValueQueryData.class.getName(),
+                ListQueryData.class.getName(),
```

Comment:
Is this right? I wasn't sure but it seems to work.

Reply:
I don't really know how this works. If this change is removed, does something break?

Reply:
yes, looks good. Serialization engine needs all implementations of `Externalizable` to be added here to be able to [instantiate classes](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/javarosa/core/util/externalizable/PrototypeFactory.java#L135) correctly while deserializing.